File: 13-10-2023_Floor_In_BST.java
class Solution {
    public static int floor(Node root, int x){
        if(root == null){
            return -1;
        }
        int temp = -1;
        int present = Integer.MIN_VALUE;
        if(root.data > x){
            temp = floor(root.left, x);
        } else if(root.data <= x){
            present = root.data;
            temp = floor(root.right, x);
        }
        return Math.max(present, temp);
    }
}

File: 13-10-2023_Floor_In_BST.cpp
class Solution{
public:
    int floor(Node* root, int x){
        if(root == NULL){
            return -1;
        }
        int temp, present = INT_MIN;
        if(root -> data > x){
            temp = floor(root -> left, x);
        }else if(root -> data <= x){
            present = root -> data;
            temp = floor(root -> right, x);
        }
        return max(present, temp);
    }
};
